ciao-image/datastore/cache.go

```diff
@@ -45,6 +45,7 @@ func (c *ImageCache) CreateImage(i Image) error {
 	c.lock.Lock()
 
 	c.images[i.ID] = i
+	c.metaDs.Write(i)
```

Comment:
Does Write return an error. If so we should trap it and maybe try to write before we update the map.  This way we have nothing to roll back.